url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/manager/panel/PartsManagerPanelV2.java

```diff
@@ -0,0 +1,417 @@
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import javax.swing.BorderFactory;
+import javax.swing.BoxLayout;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JSlider;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+
+import manager.PartsManager;
+import manager.util.ClickablePanel;
+import manager.util.ClickablePanelClickHandler;
+import manager.util.OverlayPanel;
+import manager.util.WhiteLabel;
+import Utils.Constants;
+import factory.PartType;
+
+/**
+ * Prettified PartsManagerPanel, powers PartsManager. Uses PartsListPanel to
+ * display parts.
+ * 
+ * @author Peter Zhang
+ */
+public class PartsManagerPanelV2 extends JPanel {
+
+	private JPanel panels;
+	private OverlayPanel leftPanel;
+	private JScrollPane jsp;
+	private PartsListPanel rightPanel;
+
+	private WhiteLabel leftTitle;
+	private JTextField nameField;
+	private WhiteLabel numField;
+	private JTextArea descField;
+	private JButton submitButton;
+
+	private JSlider badChanceScroller;
+	private static final int CHANCE_MIN = 0;
+	private static final int CHANCE_MAX = 100;
+	private static final int CHANCE_INIT = 0;
+
+	private PartsManager manager;
+
+	private boolean isEditing;
+	private boolean isDeleting;
+
+	private JPanel imagePanel;
+	private WhiteLabel imageLabel;
+	private JScrollPane imageSelect;
+	private JPanel imageSelectPanel;
+	private HashMap<String, ClickablePanel> imageClickablePanels = new HashMap<String, ClickablePanel>();
+	private String selectedImagePath = new String("1");
+
+	public PartsManagerPanelV2(PartsManager mngr) {
+		manager = mngr;
+
+		setLayout(new BorderLayout());
+		setBorder(Constants.PADDING);
+
+		JLabel title = new WhiteLabel("Parts Manager");
+		title.setFont(new Font("Arial", Font.BOLD, 30));
+		title.setBorder(Constants.VERTICAL_PADDING);
+		add(title, BorderLayout.NORTH);
+
+		panels = new JPanel(new GridLayout(1, 2));
+		panels.setOpaque(false);
+		panels.setVisible(true);
+		add(panels);
+
+		leftPanel = new OverlayPanel();
+		leftPanel.setVisible(true);
+		panels.add(leftPanel);
+
+		rightPanel = new PartsListPanel(
+				new PartsListPanel.PartsListPanelHandler() {
+					@Override
+					public void panelClicked(PartType pt) {
+						startEditing(pt);
+					}
+
+					@Override
+					public void buttonClicked(PartType pt) {
+						startDeleting(pt);
+					}
+				});
+		rightPanel.setVisible(true);
+		rightPanel.setBackground(new Color(0, 0, 0, 30));
+
+		jsp = new JScrollPane(rightPanel,
+				JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+				JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+		jsp.setOpaque(false);
+		jsp.getViewport().setOpaque(false);
+		panels.add(jsp);
+
+		setUpLeftPanel();
+	}
+
+	public void setUpLeftPanel() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 108 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks such as on lines 128-144, lines 146-161, lines 163-180, lines 182-203, lines 205-251 into separate methods.